torch/csrc/dynamo/compiled_autograd.h

```diff
@@ -236,6 +252,41 @@ class CompiledNodeArgs {
     collect(t.first);
     collect(t.second);
   }
+  template <typename V>
+  void collect(const ska::flat_hash_map<std::string, V>& m) {
+    collect_size(m.size());
+
+    std::vector<std::string> keys;
+    keys.reserve(m.size());
+    std::transform(
+        m.begin(), m.end(), std::back_inserter(keys), [](const auto& entry) {
+          return entry.first;
+        });
+    // stable sort to ensure deterministic collect order
+    std::stable_sort(keys.begin(), keys.end());
+    for (const auto& k : keys) {
+      collect(k);
+      collect(m.at(k));
+    }
+  }
+  void collect(const at::IValue& iv) {
+    if (iv.isList()) {
+      c10::List<at::IValue> list = iv.toList();
+      collect_size(list.size());
+      for (const auto& i : list) {
+        collect(i);
+      }
+    } else if (iv.isGenericDict()) {
+      c10::Dict<at::IValue, at::IValue> ordered_dict = iv.toGenericDict();
+      collect_size(ordered_dict.size());
+      for (const auto& entry : ordered_dict) {
+        collect(entry.key());
+        collect(entry.value());
+      }
+    } else {
+      collect(at::IValue::hash(iv));
```

Comment:
What other types hit this case?  Are they all hashable?

Reply:
These are the non-hashable ivalue types:
https://github.com/pytorch/pytorch/blob/368f242e377108a29db9ae821126a9df4b6e40a0/aten/src/ATen/core/ivalue.cpp#L394-L410

From my inspection of internal autograd::Function code, most just need tensors, primitives and list/dict

Reply:
Perhaps just turn this into an assert then.  Many of these will be wrong.